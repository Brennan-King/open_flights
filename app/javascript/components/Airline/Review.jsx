import React from 'react';
import PropTypes from 'prop-types';
import AirlineRating from '../AirlineRating/AirlineRating';
import DeleteReviewButton from './DeleteReviewButton';
import reviewIcon from '../../images/reviewIcon.png';
import './Review.scss';

const propTypes = {
  /** The title of the review */
  reviewTitle: PropTypes.string,
  /** The description of the review */
  reviewDescription: PropTypes.string,
  /** The score of the review */
  reviewScore: PropTypes.number,
  /** The id of the review */
  reviewId: PropTypes.number,
  /** Callback function to set delete button state */
  setDeleteButtonClicked: PropTypes.func,
};

const propDefaults = {
  reviewTitle: '',
  reviewDescription: '',
  reviewScore: 0,
  reviewId: 0,
  setDeleteButtonClicked: () => {},
};

/**
 * Constructs a review component used to display a specific review.
 * @param {object} props the props used to configure the review
 * @return {node} a div containing the configured review
 */
const Review = (props) => {
  const {reviewTitle,
    reviewDescription,
    reviewScore,
    reviewId,
    setDeleteButtonClicked,
  } = props;

  return (
    <div className="review-card">
      <div className="review-rating-container">
        <img
          className="review-icon"
          src={reviewIcon}
          alt="reviewIcon"
        />
        <AirlineRating
          reviewScore={reviewScore}
        />
        <DeleteReviewButton
          reviewId={reviewId}
          setDeleteButtonClicked={setDeleteButtonClicked}
        />
      </div>
      <div className="review-title">{reviewTitle}</div>
      <div className="review-description">{reviewDescription}</div>
    </div>
  );
};

Review.propTypes = propTypes;
Review.propDefaults = propDefaults;
export default Review;
