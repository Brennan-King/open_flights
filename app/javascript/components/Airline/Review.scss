.review-card {
  border: 2px solid black;
  border-radius: 4px;
  padding: 20px;
  margin: 0 0 20px 0;
  box-shadow: 7px 7px 7px lightgray;
}

.review-rating-container {
  display: flex;
  flex-direction: row;
}

.review-title {
  padding: 20px 0 0 0;
  font-size: 18px;
}

.review-description {
  padding: 0 0 20px 0;
  font-size: 14px;
}