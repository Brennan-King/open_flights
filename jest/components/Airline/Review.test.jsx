/* eslint-disable max-len */
import 'jsdom-global/register';
import React from 'react';
import Review from '../../../app/javascript/components/Airline/Review';
import {configure, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({adapter: new Adapter()});

/**
 * Test suite for Review component
 */
describe('Review', () => {
  it('should render a Review component with default props', () => {
    const component = mount(
        <Review />,
    );

    expect(component).toMatchSnapshot();
  });

  it('should render a Review component', () => {
    const component = mount(
        <Review
          reviewTitle="testReviewTitle"
          reviewDescription="testReviewDescription"
          reviewScore={5}
          reviewId={1}
          setDeleteButtonClicked={()=>{}}
        />,
    );

    expect(component).toMatchSnapshot();
  });
});
