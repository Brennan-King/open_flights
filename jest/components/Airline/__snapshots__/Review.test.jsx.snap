// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review should render a Review component 1`] = `
<Review
  reviewDescription="testReviewDescription"
  reviewId={1}
  reviewScore={5}
  reviewTitle="testReviewTitle"
  setDeleteButtonClicked={[Function]}
>
  <div
    className="review-card"
  >
    <div
      className="review-rating-container"
    >
      <img
        alt="reviewIcon"
        className="review-icon"
        src="test-file-stub"
      />
      <AirlineRating
        reviewScore={5}
      >
        <span
          className="airline-rating-wrapper"
        >
          <div
            className="airline-rating-filled-in"
          />
          <div
            className="airline-rating-filled-in"
          />
          <div
            className="airline-rating-filled-in"
          />
          <div
            className="airline-rating-filled-in"
          />
          <div
            className="airline-rating-filled-in"
          />
        </span>
      </AirlineRating>
      <DeleteReviewButton
        reviewId={1}
        setDeleteButtonClicked={[Function]}
      >
        <div
          className="delete-review-button"
          id="delete-review-button"
          onClick={[Function]}
        >
          <div
            className="x-symbol"
          >
            x
          </div>
        </div>
      </DeleteReviewButton>
    </div>
    <div
      className="review-title"
    >
      testReviewTitle
    </div>
    <div
      className="review-description"
    >
      testReviewDescription
    </div>
  </div>
</Review>
`;

exports[`Review should render a Review component with default props 1`] = `
<Review
  reviewDescription=""
  reviewId={0}
  reviewScore={0}
  reviewTitle=""
  setDeleteButtonClicked={[Function]}
>
  <div
    className="review-card"
  >
    <div
      className="review-rating-container"
    >
      <img
        alt="reviewIcon"
        className="review-icon"
        src="test-file-stub"
      />
      <AirlineRating
        reviewScore={0}
      >
        <span
          className="airline-rating-wrapper"
        >
          <div
            className="airline-rating-not-filled-in"
          />
          <div
            className="airline-rating-not-filled-in"
          />
          <div
            className="airline-rating-not-filled-in"
          />
          <div
            className="airline-rating-not-filled-in"
          />
          <div
            className="airline-rating-not-filled-in"
          />
        </span>
      </AirlineRating>
      <DeleteReviewButton
        reviewId={0}
        setDeleteButtonClicked={[Function]}
      >
        <div
          className="delete-review-button"
          id="delete-review-button"
          onClick={[Function]}
        >
          <div
            className="x-symbol"
          >
            x
          </div>
        </div>
      </DeleteReviewButton>
    </div>
    <div
      className="review-title"
    />
    <div
      className="review-description"
    />
  </div>
</Review>
`;
